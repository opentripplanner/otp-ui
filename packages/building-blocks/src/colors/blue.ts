const blue = {
  50: "#E5ECF3",
  100: "#CCDAE7",
  200: "#B2C7DB",
  300: "#99B5CF",
  400: "#7FA2C2",
  500: "#6690B7",
  600: "#4D7EAB",
  // 700 and above text is WCAG AA accessible on white backgrounds.
  700: "#336B9E",
  // 800 and above text is WCAG AAA accessible on white backgrounds.
  800: "#1A5992",
  900: "#004686"
};

export default blue;
