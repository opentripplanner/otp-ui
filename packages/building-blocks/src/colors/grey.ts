const grey = {
  50: "#ECECEC",
  100: "#D9D9D9",
  200: "#C5C5C5",
  300: "#B3B3B3",
  400: "#9F9F9F",
  500: "#8C8C8C",
  600: "#797979",
  // 700 and above text is WCAG AA accessible on white backgrounds.
  700: "#666666",
  // 800 and above text is WCAG AAA accessible on white backgrounds.
  800: "#535353",
  900: "#404040"
};

export default grey;
