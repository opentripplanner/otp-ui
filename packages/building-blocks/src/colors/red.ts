const red = {
  50: "#F7E7E6",
  100: "#EFCFCE",
  200: "#E7B6B4",
  300: "#DF9E9C",
  400: "#D78683",
  500: "#CF6E6B",
  600: "#C75652",
  // 700 and above text is WCAG AA accessible on white backgrounds.
  700: "#BF3E3A",
  800: "#B72620",
  // 900 text is WCAG AAA accessible on white backgrounds.
  900: "#AF0E08"
};

export default red;
