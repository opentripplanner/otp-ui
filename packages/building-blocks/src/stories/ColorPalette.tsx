import React from "react";

import styled from "styled-components";
import { getMostReadableTextColor } from "@opentripplanner/core-utils/lib/route";

export interface Hue {
  [key: number]: string;
}

interface Props {
  color: Hue;
}

const ColorBlock = styled.div<{ hex: string }>`
  align-items: center;
  background-color: ${props => props.hex};
  color: ${props => getMostReadableTextColor(props.hex, "#ffffff")};
  text-shadow: ${props =>
    getMostReadableTextColor(props.hex, "#ffffff") === "#ffffff"
      ? "1px 1px 2px black"
      : ""};
  display: flex;
  font-family: Arial, Helvetica, sans-serif;
  height: 40px;
  justify-content: space-between;
  margin: 0;
  padding: 10px;
  width: 300px;

  &:first-of-type {
    border-radius: 8px 8px 0 0;
  }
  &:last-of-type {
    border-radius: 0 0 8px 8px;
  }
`;

const ColorPalette = ({ color }: Props): JSX.Element => {
  const colorArray: Array<Hue> = Object.entries(color);

  return (
    <>
      {colorArray.map(
        (hue: Hue): JSX.Element => {
          return (
            <ColorBlock key={hue[0]} hex={hue[1]}>
              <p>{hue[0]}</p>
              <p>{hue[1]}</p>
            </ColorBlock>
          );
        }
      )}
    </>
  );
};
export default ColorPalette;
