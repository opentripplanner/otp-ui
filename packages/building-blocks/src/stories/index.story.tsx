import React, { ReactElement } from "react";
import ColorPalette from "./ColorPalette";

import grey from "../colors/grey";
import blue from "../colors/blue";
import red from "../colors/red";

export default {
  title: "Building-Blocks/Colors"
};

export const Grey = (): ReactElement => {
  return <ColorPalette color={grey} />;
};

export const Blue = (): ReactElement => {
  return <ColorPalette color={blue} />;
};

export const Red = (): ReactElement => {
  return <ColorPalette color={red} />;
};
