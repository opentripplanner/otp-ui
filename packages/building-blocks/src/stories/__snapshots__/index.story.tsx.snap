// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Building-Blocks/Colors Blue smoke-test 1`] = `
<div class="ColorPalette__ColorBlock-sc-fil65c-0 bFLnsf">
  <p>
    50
  </p>
  <p>
    #E5ECF3
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 fslHeP">
  <p>
    100
  </p>
  <p>
    #CCDAE7
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 hYrXqS">
  <p>
    200
  </p>
  <p>
    #B2C7DB
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 cwDhpW">
  <p>
    300
  </p>
  <p>
    #99B5CF
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 kpjxeq">
  <p>
    400
  </p>
  <p>
    #7FA2C2
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 cIdVEx">
  <p>
    500
  </p>
  <p>
    #6690B7
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 YgYzo">
  <p>
    600
  </p>
  <p>
    #4D7EAB
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 ixMEWR">
  <p>
    700
  </p>
  <p>
    #336B9E
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 bwAoty">
  <p>
    800
  </p>
  <p>
    #1A5992
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 dqaynR">
  <p>
    900
  </p>
  <p>
    #004686
  </p>
</div>
`;

exports[`Building-Blocks/Colors Grey smoke-test 1`] = `
<div class="ColorPalette__ColorBlock-sc-fil65c-0 fQVZiO">
  <p>
    50
  </p>
  <p>
    #ECECEC
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 oxwmZ">
  <p>
    100
  </p>
  <p>
    #D9D9D9
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 jueztC">
  <p>
    200
  </p>
  <p>
    #C5C5C5
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 ipjnjR">
  <p>
    300
  </p>
  <p>
    #B3B3B3
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 KHVkK">
  <p>
    400
  </p>
  <p>
    #9F9F9F
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 dtcMei">
  <p>
    500
  </p>
  <p>
    #8C8C8C
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 iKIpwf">
  <p>
    600
  </p>
  <p>
    #797979
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 foMzhh">
  <p>
    700
  </p>
  <p>
    #666666
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 efnWWL">
  <p>
    800
  </p>
  <p>
    #535353
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 hKGOgp">
  <p>
    900
  </p>
  <p>
    #404040
  </p>
</div>
`;

exports[`Building-Blocks/Colors Red smoke-test 1`] = `
<div class="ColorPalette__ColorBlock-sc-fil65c-0 eLDsHA">
  <p>
    50
  </p>
  <p>
    #F7E7E6
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 bwwaXk">
  <p>
    100
  </p>
  <p>
    #EFCFCE
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 jTuSFc">
  <p>
    200
  </p>
  <p>
    #E7B6B4
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 lnSWIY">
  <p>
    300
  </p>
  <p>
    #DF9E9C
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 jeVJXr">
  <p>
    400
  </p>
  <p>
    #D78683
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 dDTLvn">
  <p>
    500
  </p>
  <p>
    #CF6E6B
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 jPUfgl">
  <p>
    600
  </p>
  <p>
    #C75652
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 jjxBwB">
  <p>
    700
  </p>
  <p>
    #BF3E3A
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 gAlgpe">
  <p>
    800
  </p>
  <p>
    #B72620
  </p>
</div>
<div class="ColorPalette__ColorBlock-sc-fil65c-0 lewmhT">
  <p>
    900
  </p>
  <p>
    #AF0E08
  </p>
</div>
`;
